code/modules/surgery/organs/liver.dm

```diff
@@ -25,22 +25,17 @@
 			//slowly heal liver damage
 			damage = max(0, damage - 0.1)
 
-			if(filterToxins)
+			if(filterToxins && !owner.has_trait(TRAIT_TOXINLOVER))
 				//handle liver toxin filtration
-				var/toxamount
 				var/static/list/toxinstypecache = typecacheof(/datum/reagent/toxin)
 				for(var/I in C.reagents.reagent_list)
 					var/datum/reagent/pickedreagent = I
 					if(is_type_in_typecache(pickedreagent, toxinstypecache))
-						toxamount += C.reagents.get_reagent_amount(initial(pickedreagent.id))
-				if(toxamount <= toxTolerance && toxamount > 0)
-					for(var/I in C.reagents.reagent_list)
-						var/datum/reagent/pickedreagent = I
-						if(is_type_in_typecache(pickedreagent, toxinstypecache))
+						var/thisamount = C.reagents.get_reagent_amount(initial(pickedreagent.id))

```

Comment:
is pickedreagent a typepath? looks like an instance from `C.reagents.reagent_list`

Reply:
it's an instance but `initial(thing.id)` is faster for instance-invariant data that has to be read from the type

Reply:
I hate byond...